go-controller/pkg/factory/factory.go

```diff
@@ -41,25 +41,47 @@ func (factory *Factory) CreateOvnController() *ovn.Controller {
 	namespaceInformer := factory.IFactory.Core().V1().Namespaces()
 
 	return &ovn.Controller{
-		StartPodWatch: func(handler cache.ResourceEventHandler) {
-			podInformer.Informer().AddEventHandler(handler)
+		StartPodWatch: func(handler cache.ResourceEventHandler, processExisting func([]interface{})) {
 			go podInformer.Informer().Run(utilwait.NeverStop)
+			cache.WaitForCacheSync(utilwait.NeverStop, podInformer.Informer().HasSynced)
```

Comment:
WaitForCacheSync() can fail, may be we should log an error or pass the return value to processExisting callback so that it can decide what to do (it may want to perform an op only in case of successful cache sync).